[modernize][Videoplayer] Prefer 'override' usage

hudokkow opened the pull request:
Follow up of https://github.com/garbear/xbmc/pull/83

@garbear, @FernetMenta

hudokkow committed:
[modernize][Videoplayer] Prefer 'override' usage

garbear commented:
I'll follow up with some changes from the comments made in #12250

FernetMenta commented on xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemux.h:
```diff
@@ -243,29 +243,29 @@ class CDVDDemux
   /*
    * Reset the entire demuxer (same result as closing and opening it)
    */
-  virtual void Reset() = 0;
```
I already mentioned that this is wrong. please don't change behaviour in such kind of bulk cosmetic changes.

hudokkow replied:
Think this was some sort of rebase stupidity and lack of attention from my part. Oddly, after rebasing today it went away by itself.

FernetMenta commented:
do not hide architectural changes into this kind of PRs. that is the reason why #12250 got stuck

garbear commented:
@hudokkow please remove my comment from your PR. I sent the updated version to https://github.com/hudokkow/xbmc/pull/1

garbear commented:
@FernetMenta I have addressed the architectural comments from #12250: https://github.com/hudokkow/xbmc/pull/1/files

Should I send to this PR, or wait for this PR to be merged (without any architectural changes) and then PR to `vpupdates`?

FernetMenta commented:
please focus only on "override" changes in this PR. this is what the title says. every thing else we should do in separeate PRs.
btw @hudokkow thanks for doing this on this branch.

hudokkow commented:
Rebased and cleaned/fixed up a lot of vertical alignment stuff.

Better review with https://github.com/xbmc/xbmc/pull/12440/files?w=1 to ignore tab/space changes.

hudokkow commented:
jenkins build this please

FernetMenta commented:
thanks
